texar/torch/modules/pretrained/pretrained_bert.py

```diff
@@ -231,3 +277,78 @@ def _init_from_checkpoint(self, cache_dir: str, **kwargs):
                 else:
                     raise NameError(f"Variable with name '{name}' not found")
                 idx += 1
+
+    def _init_roberta_from_checkpoint(self, cache_dir: str):
+        global_tensor_map = {
+            'decoder.sentence_encoder.embed_tokens.weight':
+                'word_embedder._embedding',
+            'decoder.sentence_encoder.embed_positions.weight':
+                'position_embedder._embedding',
+            'decoder.sentence_encoder.emb_layer_norm.weight':
+                'encoder.input_normalizer.weight',
+            'decoder.sentence_encoder.emb_layer_norm.bias':
+                'encoder.input_normalizer.bias',
+        }
+
+        attention_tensor_map = {
+            'final_layer_norm.weight':
+                'encoder.output_layer_norm.{}.weight',
+            'final_layer_norm.bias':
+                'encoder.output_layer_norm.{}.bias',
+            'fc1.weight':
+                'encoder.poswise_networks.{}._layers.0.weight',
+            'fc1.bias':
+                'encoder.poswise_networks.{}._layers.0.bias',
+            'fc2.weight':
+                'encoder.poswise_networks.{}._layers.2.weight',
+            'fc2.bias':
+                'encoder.poswise_networks.{}._layers.2.bias',
+            'self_attn_layer_norm.weight':
+                'encoder.poswise_layer_norm.{}.weight',
+            'self_attn_layer_norm.bias':
+                'encoder.poswise_layer_norm.{}.bias',
+            'self_attn.out_proj.weight':
+                'encoder.self_attns.{}.O_dense.weight',
+            'self_attn.out_proj.bias':
+                'encoder.self_attns.{}.O_dense.bias',
+            'self_attn.in_proj_weight': [
+                'encoder.self_attns.{}.Q_dense.weight',
+                'encoder.self_attns.{}.K_dense.weight',
+                'encoder.self_attns.{}.V_dense.weight',
+            ],
+            'self_attn.in_proj_bias': [
+                'encoder.self_attns.{}.Q_dense.bias',
+                'encoder.self_attns.{}.K_dense.bias',
+                'encoder.self_attns.{}.V_dense.bias'
+            ],
+        }
+
+        checkpoint_path = os.path.abspath(os.path.join(cache_dir, 'model.pt'))
+        device = torch.device(
```

Comment:
This has potential issues because the model (`self`) might not be on the device as defined here. For robustness we can fetch a parameter from the model and use its parameter, probably like this:
```python
device = next(self.parameters()).device
```